pipeline/pipeline/backend.py

```diff
@@ -143,48 +143,31 @@ def run(self, pipeline, dry_run, verbose, delete_scratch_on_exit):  # pylint: di
         task_to_job_mapping = {}
         job_id_to_command = {}
 
-        volumes = [{'volume': {'name': 'pipeline-test-0-1--hail-is-service-account-key',
-                               'secret': {'optional': False,
-                                          'secretName': 'pipeline-test-0-1--hail-is-service-account-key'}},
-                    'volume_mount': {'mountPath': '/secrets',
-                                     'name': 'pipeline-test-0-1--hail-is-service-account-key',
-                                     'readOnly': True}}]
-
-        activate_service_account = 'gcloud auth activate-service-account ' \
-                                   'pipeline-test-0-1--hail-is@hail-vdc.iam.gserviceaccount.com ' \
-                                   '--key-file /secrets/pipeline-test-0-1--hail-is.key'
-
         for task in pipeline._tasks:
             def copy_input(r):
                 if isinstance(r, ResourceFile):
                     if isinstance(r, InputResourceFile):
-                        return [f'gsutil cp {r._input_path} {r._get_path(local_tmpdir)}']
+                        return [(r._input_path, r._get_path(local_tmpdir))]
                     else:
                         assert isinstance(r, TaskResourceFile)
-                        return [f'gsutil cp {r._get_path(remote_tmpdir)} {r._get_path(local_tmpdir)}']
+                        return [(r._get_path(remote_tmpdir), r._get_path(local_tmpdir))]
                 else:
                     assert isinstance(r, ResourceGroup)
                     return [x for _, rf in r._resources.items() for x in copy_input(rf)]
 
             def copy_output(r):
                 assert r._source == task  # pylint: disable-msg=W0640
                 if isinstance(r, TaskResourceFile):
-                    return [f'gsutil cp {r._get_path(local_tmpdir)} {r._get_path(remote_tmpdir)}']
+                    return [
+                        (r._get_path(local_tmpdir), r._get_path(remote_tmpdir)),
+                        *[(r._get_path(local_tmpdir), output_path) for output_path in r._output_paths]]
```

Comment:
This changes the behavior of my original code. You can write out an input resource file, but your code won't write them out because it's only looking at the task outputs. That's why I had the code at the end that went through the entire map of a pipeline's resources and wrote everything out at once. Perhaps adding an additional data structure that looks for input files will solve this. I also had code to handle a resource group and writing the file with the extension being the identifier of the resource group's file. Example `p.write_output(t.bfile, 'gs://example') should write out gs://example.bed, gs://example.bim, and gs://example.fam`

Also, should probably add asserts that source != destination when copying in Batch.